azurelinuxagent/common/utils/archive.py

```diff
@@ -208,25 +208,15 @@ def __init__(self, timestamp, tag=None):
         self._root = os.path.join(conf.get_lib_dir(), ARCHIVE_DIRECTORY_NAME, "{0}_{1}".format(timestamp, tag) if tag is not None else timestamp)
 
     def save(self, data, file_name):
-        def write_to_file(d, f):
-            with open(f, "w") as h:
-                h.write(d)
-
-        self._save(write_to_file, data, file_name)
-
-    def _save_file(self, source_file, target_name):
-        self._save(shutil.move, source_file, target_name)
-
-    def _save(self, function, source, target_name):
         try:
             if not os.path.exists(self._root):
                 fileutil.mkdir(self._root, mode=0o700)
-            target = os.path.join(self._root, target_name)
-            function(source, target)
+            with open(os.path.join(self._root, file_name), "w") as handle:
+                handle.write(data)
         except Exception as e:
             if not self._errors:  # report only 1 error per directory
                 self._errors = True
-                logger.warn("Failed to save goal state file {0}: {1} [no additional errors saving the goal state will be reported]".format(target_name, e))
+                logger.warn("Failed to save goal state file {0}: {1} [no additional errors saving the goal state will be reported]".format(file_name, e))
```

Comment:
Maybe this error message needs updating? `Failed to save {file_name} for incarnation {id} [no additional errors saving the goal state will be reported]: {error}`

Reply:
i see... by "goal state file" i did not mean GoalState.xml, but a file related to the goal state... i'll update this to "Failed to save {0} to the goal state history: {1}" (the incarnation or etag are already logged 1 or 2 lines above this message)